sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileClient.java

```diff
@@ -724,9 +1008,47 @@ public Response<ShareFileProperties> downloadToFileWithResponse(String downloadF
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<ShareFileProperties> downloadToFileWithResponse(String downloadFilePath, ShareFileRange range,
         ShareRequestConditions requestConditions, Duration timeout, Context context) {
-        Mono<Response<ShareFileProperties>> response = shareFileAsyncClient.downloadToFileWithResponse(downloadFilePath,
-            range, requestConditions, context);
-        return StorageImplUtils.blockWithOptionalTimeout(response, timeout);
+        Objects.requireNonNull(downloadFilePath, "'downloadFilePath' cannot be null.");
+
+        // Setup initial conditions and file channel
+        ShareRequestConditions finalRequestConditions = requestConditions == null ? new ShareRequestConditions()
+            : requestConditions;
+        //ShareFileRange finalRange = range;
+        try (FileChannel fileChannel = FileChannel.open(Paths.get(downloadFilePath), StandardOpenOption.CREATE,
+            StandardOpenOption.WRITE)) {
```

Comment:
nit: Not related to this PR but something I noticed.

We should allow for the file open options to be configured similarly to how Blob allows them to be set for downloadToFile